packages/swingset-liveslots/src/virtualReferences.js

```diff
@@ -636,7 +650,8 @@ export function makeVirtualReferenceManager(
     }
   }
 
-  function finalizeDroppedCollection(descriptor) {
+  function finalizeDroppedCollection({ descriptor }) {
+    // the 'wr' WeakRef will be dropped about now
```

Comment:
So little tangent on what optimizations engines are allowed or not. Technically, since `wr` is never observed by the program, the JS engine is allowed to not keep it in the `heldValue`, which means that the WeakRef would actually not exist in the first place.

If we wanted to be as close to the spec rules as possible, we would need to actually try to deref the weakref and assert it's empty (or even better, log the target if not empty, causing an actual observation). Obviously no engine goes that far in its optimizations, certainly not XS, and the destructuring dropping the wr is in fact completely sufficient.

Reply:
wow, the engine is allowed to inspect the callback's code to determine that? hardcore.

Reply:
The engine is allowed to do anything as long as it's not observable to the program. If a property of an object is never used, it doesn't have to store it ;) But in practice engines are not omniscient enough.